src/persist/pnew.hpp

```diff
@@ -0,0 +1,14 @@
+// NOTE: don't include this file elsewhere!
+// this is supposed to be a part of Recoverable.hpp
+
+// TODO: replace `new` operator of T with
+// per-heap allocation and placement new.
+template <typename T, typename... Types> 
+T* pnew(Types... args) 
+{ 
+    T* ret = new T(args...);
```

Comment:
I optimized a little here in commit [204c67](https://github.com/urcs-sync/Montage/pull/3/commits/204c67fccfe7b0628c477d4b8b48d22ebb4d524c). Hope I got your idea correctly.

Reply:
Oh I don't know we can do that... Thanks a lot! I think I can put this back to `Recoverable.hpp` and delete this file.

Reply:
Nice.